salt/states/pip_state.py

```diff
@@ -163,8 +164,11 @@ def _check_pkg_version_format(pkg):
         ret['version_spec'] = []
     else:
         ret['result'] = True
-        ret['prefix'] = install_req.req.project_name
-        ret['version_spec'] = install_req.req.specs
+        if hasattr(install_req.req, 'project_name'):
+            ret['prefix'] = install_req.req.project_name
+        else:
+            ret['prefix'] = re.sub('[^A-Za-z0-9.]+', '-', install_req.req.name)
+        ret['version_spec'] = [(spec.operator, spec.version) for spec in install_req.req.specifier]
```

Comment:
Yea, if there's something pip can actually stablize and make promises about, that's an option too. Our standard answer to "can I import pip" is basically "we don't support it, but if there's something you need let us know and we'll see if we can make something work".